bootstrap/controllers/orchestrated_inplace_upgrade_controller.go

```diff
@@ -91,41 +87,36 @@ func (r *MachineDeploymentReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		return ctrl.Result{}, fmt.Errorf("failed to get MachineDeployment: %w", err)
 	}
 
-	if r.getUpgradeInstructions(machineDeployment) == "" {
+	if inplace.GetUpgradeInstructions(machineDeployment) == "" {
 		log.V(1).Info("MachineDeployment has no upgrade instructions, skipping reconciliation")
 		return ctrl.Result{}, nil
 	}
 
-	if !machineDeployment.DeletionTimestamp.IsZero() {
+	if isDeleted(machineDeployment) {
 		log.V(1).Info("MachineDeployment is being deleted, skipping reconciliation")
 		return ctrl.Result{}, nil
 	}
 
-	ownedMachines, err := r.getOwnedMachines(ctx, machineDeployment)
-	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("failed to get owned machines: %w", err)
-	}
-
-	scope, err := r.createScope(machineDeployment, ownedMachines)
+	scope, err := r.createScope(ctx, machineDeployment)
 	if err != nil {
 		return ctrl.Result{}, fmt.Errorf("failed to create scope: %w", err)
 	}
 
 	// Starting the upgrade process
 	var upgradedMachines int
-	for _, m := range ownedMachines {
-		if r.isMachineUpgraded(scope, m) {
+	for _, m := range scope.ownedMachines {
+		if inplace.IsUpgraded(m, scope.upgradeTo) {
```

Comment:
Out of curiosity, why do we go from ownedMachines to scope.ownedMachines?

Reply:
Just so that there's a single reference/thing-we-can-use. Before, we declared an `ownedMachines` variable and passed it to the `createScope` function. IMHO that didn't really help with dependency injection and I believe we're good to let the `createScope` to also create the `ownedMachines` in itself. So now, we only have the `ownedMachines` through the `scope.owedMachines`.